quartz/plugins/transformers/lastmod.ts

```diff
@@ -49,12 +59,16 @@ export const CreatedModifiedDate: QuartzTransformerPlugin<Partial<Options> | und
                 const st = await fs.promises.stat(fullFp)
                 created ||= st.birthtimeMs
                 modified ||= st.mtimeMs
-              } else if (source === "frontmatter" && file.data.frontmatter) {
-                created ||= file.data.frontmatter.date as MaybeDate
-                modified ||= file.data.frontmatter.lastmod as MaybeDate
-                modified ||= file.data.frontmatter.updated as MaybeDate
-                modified ||= file.data.frontmatter["last-modified"] as MaybeDate
-                published ||= file.data.frontmatter.publishDate as MaybeDate
+              } else if (source === "frontmatter" && opts.properties && file.data.frontmatter) {
```

Comment:
we should explicitly spread in the default options for the properties subfield on L43 here and get rid of the nullish assertion `!`

Reply:
I'm not sure how to accomplish this.

I've tried a few ways, but I'm still receiving the possibly undefined error.

```ts
const opts = { ...defaultOptions, properties: ...defaultOptions.properties, ...userOpts }
```

 ```ts
const opts = {
    ...defaultOptions,
    properties: {
      created: defaultOptions.properties?.created as string[],
      modified: defaultOptions.properties?.created as string[],
      published: defaultOptions.properties?.published as string[],
    },
    ...userOpts,
  }
```

![image](https://github.com/jackyzha0/quartz/assets/161913/f267ba6e-fcbb-4a78-960c-ec91d0d09b13)

Reply:
Still complaining.

The only way I've been able to get rid of the error without the null assertion is by
```ts
if (opts.properties.created) {
  for (const createdProperty of opts.properties.created) {
    created ||= file.data.frontmatter[createdProperty] as MaybeDate
  }
}
```

Which is lame, because it's always true. It sounds like a typescript bug to me.

Reply:
hmm just nit thing:

since you are adding a lot of custom options here, should we use `switch` case for this instead of if loops?